libwasmvm/src/api.rs

```diff
@@ -254,8 +261,71 @@ impl BackendApi for GoApi {
             Err(e) => return (Err(BackendError::unknown(e.to_string())), gas_info),
         };
         callee_instance.env.set_serialized_env(&contract_env);
-        callee_instance.set_storage_readonly(caller_env.is_storage_readonly());
         gas_info.cost += instantiate_cost;
+        // set the callee_instance to read-only before calling _list_callable_points
+        // This is because it is possible to define read-write functions in the _list_callable_points function
```

Comment:
This Function `fn contract_call<A, S, Q>` processing is large. It's hard to understand what the code means.
How about creating one function that takes callee_instance as input and sets permissions?

Reply:
The code is already long before I make my changes, will this be resolved by creating an external function?
Also, since the code has explanatory text, I think it would not be hard to understand what the code means.